Add style guide to Contribute to the docs (#4060)
elastic/docs-content: contribute-docs/style-guide/seo.md

> # SEO guidelines

Reviewer: What's good for SEO is basically what's good for a human reader too, I wonder if shouldn't subtly reorient this towards "Best practices for structuring your pages". We should all be referring to this and absorbing it into our muscle memory over time!

I think this page is one of the most useful and immediately practical for all docs contributors, so I wonder if it might be buried and lost a bit in the style guide section.

To my eyes, ideally it would be a sibling to how to create good issues and how to write cumulative docs. 🤔 

But maybe that's something we can think about in a follow-up.

Best practices:

* The meta title should be concise, descriptive, and include the main keyword or topic of the page.  
* It's important that the H1 accurately reflects the page's content and purpose, as this will be shown in the search results.  
* Avoid duplicating meta titles across different pages. Each should be unique.  
* Keep the H1 to a recommended length of 50–60 characters, and make sure the most important information is at the beginning.  
* The meta title is not directly visible to users on the page itself, but it's critical for search visibility and user engagement in search results.

### H2: Secondary headings

H2s break the content into logical sections, outlining key steps, concepts, or features. H2s help search engines understand subtopics and improve page scannability. Given the volume of documentation content, and the similarity between certain topics, try to make secondary headings unique.

Best practices:

* Use H2s for major sections or steps.  
* Incorporate secondary keywords where relevant.  
* H2s are descriptive so users can quickly find the information they need.

Examples:

✅Configure output settings

❌Settings

✅Deploy Filebeat to Kubernetes

❌Deploy

> ### H3s, H4s, H5s, and etc: Subheadings

Reviewer 2: ```suggestion
### Subheadings
```

* Include related terms and synonyms to improve semantic relevance.

**Organize with headings and subheadings**

* Break content into logical sections using descriptive headings (H2, H3, etc.).  
* Headings help users scan for relevant information and assist search engines in understanding content hierarchy.  
* Each heading should accurately reflect the content that follows.

**Ensure content quality, originality, and uniqueness**

* Avoid thin content—ensure each page provides substantial, unique value to the reader.  
* Do not duplicate content from other pages or sources; always tailor information for the specific context and audience.  
* Regularly review and update content to keep it accurate and relevant.

**Make content scannable**

* Use bullet points, numbered lists, and callouts for important information.  
* Highlight warnings, tips, or best practices using clear formatting.

**Provide examples and visuals**

* Include code snippets, sample commands, or screenshots to illustrate concepts.  
* Reference visuals in the text, but ensure all key information is also available in text for accessibility.

**Leverage structured data, where applicable**

> * Use structured data (e.g., [schema.org](https://schema.org/) markup) to add semantic meaning.  

Reviewer 2: ```suggestion
Use structured data (for example, [schema.org](https://schema.org/) markup) to add semantic meaning.  
```

* Structured data can help search engines display your documentation in rich results, such as FAQs or How-To snippets.  

>   Check with the documentation team for existing capabilities or implementation guidance.

Reviewer 2: This advice doesn't apply to external contributors
```suggestion

```

* Avoid using lists for layout or decorative purposes.

### Interlinking

Links are vital for both user navigation and SEO. They help users discover related content and signal content relationships to search engines. They also serve as a means for search engine crawlers to find and index pages.

Best practices:

**Use descriptive anchor text**

* Make link text meaningful (e.g., "Monitor Configuration Reference" instead of "click here").  
* Avoid vague phrases; anchor text should describe the destination or resource.

**Prioritize internal linking**

* Link to related documentation pages to guide users and strengthen site structure.  
* Use internal links to connect related documentation pages, guiding users to deeper or related topics.  
* Regularly audit and update links to avoid broken or outdated references.  
* Consider the user journey when linking, and link outside of Docs, like to the marketing site, or blog where appropriate and relevant.

**Use external links judiciously**

* Link to reputable, authoritative sources when necessary. E.g. https://www.python.org/doc/  
* Ensure external links open in a new tab/window.

**Maintain link accessibility**

> * Ensure link text is distinguishable (via color, underline, etc.).  

Reviewer 2: ```suggestion
* Ensure link text is distinguishable (for example, by using color and underline).  
```

* Avoid Overlinking  
* Don’t overload paragraphs with multiple links—prioritize the most relevant ones.

## Multimedia

Multimedia is important because it can improve user experience, site performance, and ranking signals. It can boost user engagement and retention by providing different ways to consume information.

> By adding things like image alt tags, and structured data, search engines can also pick up assets to serve them within organic search results, or use them to enhance your listing within organic search results.

Reviewer 2: ```suggestion
By adding things like image alt tags and structured data, search engines can also pick up assets to serve them within organic search results, or use them to enhance your listing within organic search results.
```

* Check with the documentation team for existing capabilities or implementation guidance. If you're an external contributor, open a GitHub issue to discuss implementation options.

## URLs

A URL (Uniform Resource Locator) is the web address of a page. Well-structured URLs benefit both SEO and user experience.

Having a URL that clearly reflects what the page is about, can increase click-through-rates within the search engine results page by matching the user query. The URL is also displayed to users in the top of a browser’s search bar. Allowing the URL to be clear, concise, and easy to read, supports a positive on-site experience.

URLs should not have special characters to ensure readability. They should also be lower-case to prevent issues with multiple URL versions that can signal duplicate content to search engines.

Best practices:
**Reflect the page topic**

* Use keywords that accurately describe the page’s subject.  
* Align the URL with the page title and H1 heading for consistency.

**Keep URLs short and readable**

* Avoid unnecessary words or parameters.  
* Use hyphens (-) to separate words for clarity.

**Use lowercase letters**

> * Always use lowercase to prevent duplicate content issues (e.g., /Guide/ vs. /guide/).

Reviewer 2: I don't think that example is up-to-date or useful
```suggestion
* Always use lowercase to prevent duplicate content issues.
```

**Avoid special characters**

* Stick to letters, numbers, and hyphens. Avoid spaces, underscores, or special symbols.

Ensure the Hierarchy is Reflected in the URL Structure

* Structure URLs to mirror the documentation’s content hierarchy, nesting child pages under their logical parent directories (e.g., /docs/solutions/security/ai/ai-assistant).  
* Ensure each child page has a valid parent URL that leads to a parent page.

## Meta keywords tag

Meta keywords are not important for SEO on Google and Bing, as they have stated that they ignore this meta tag. However, some other search engines like Yandex may still use them as a low weight ranking factor.

> It’s not necessary to add, but if needed, keywords should be kept to 5 keywords, separated by commas, while avoiding repetition, and duplication.

Reviewer 2: > It’s not necessary to add, but if needed, keywords should be kept to 5 keywords, separated by commas, while avoiding repetition, and duplication.

Do have a way to accomplish this in Docs V3? I haven't seen it mentioned in https://docs-v3-preview.elastic.dev/elastic/docs-builder/tree/main/syntax/frontmatter

* Do not deviate from template styles.  
* If new functionality is needed, discuss with the documentation team before implementing. External contributors should open a GitHub issue to propose new functionality.

**Accessible interactions**

* Make sure buttons, links, and interactive elements are easy to tap.

**Optimize media**

* Compress images and avoid large files that can slow down mobile load times.

**Avoid horizontal scrolling**

* Content should fit within the screen width without requiring users to scroll sideways.

## Content localization and SEO

Localization is not currently in scope for all Documentation content. If localization is needed there are SEO considerations that should be followed.

Best practices:

**Write in clear, simple language**

* Avoid idioms, slang, or cultural references that may not translate well.  
* Ensure that content is translated by an expert of that language.

**Consistent terminology**

* Use consistent technical terms and product names throughout the documentation.

**Customize visuals**

* Use images that are culturally appropriate and relevant to the respective language and region.  

>   * E.g., do not use English-USA language graphics if taking English-USA targeted content for a localization effort that does not use USA English.  

Reviewer 2: ```suggestion
  * For example, do not use English-USA language graphics if taking English-USA targeted content for a localization effort that does not use USA English.  
```

- [ ] No broken or outdated links  
- [ ] Links are not excessive or distracting  
- [ ] Link text is visually distinguishable & accessible

**Multimedia (images, video, etc.)**

- [ ] Visuals (images, diagrams, videos) directly support the content  
- [ ] Images are unique, high-quality, & not stock photos  
- [ ] Image file names are descriptive, lowercase, & use hyphens  
- [ ] Images are compressed/optimized for web  
- [ ] Alt text accurately describes each image’s purpose  
- [ ] No important text is embedded in images  
- [ ] Videos are relevant, compressed, & include captions/transcripts  
- [ ] Visuals are referenced in the body copy  
- [ ] Third-party visuals are properly attributed  
- [ ] Structured data is used for multimedia where possible

**URLs**

- [ ] URL is short, descriptive, & reflects the page’s topic  
- [ ] Uses lowercase letters & hyphens (no spaces, underscores, or special characters)  
- [ ] URL structure mirrors the documentation hierarchy (e.g., /docs/section/topic/)  
- [ ] Each child page has a valid parent page  
- [ ] URL matches the H1 & page focus  
- [ ] Canonical tag is set to the preferred URL

> **Meta keywords tag**

Reviewer 2: > **Meta keywords tag**

Per previous comment, is this still supported and desired?
